### ISSUE TITLE
Add JdbcRelyingPartyRegistrationRepository

### ISSUE THREAD

**Issue by sasirekha98** [spring-projects/spring-security#16012]
I have an application where the SP metadata is generated as XML and put in the database. We pull the XML and generate the authentication request. When I want to move to Spring Security, I don't find the same option where I can get the relying party details from the data source or directly provide the XML file location.

Currently, as per my understanding, we have the option to load the XML file of the IDP through metadata locations through spring security like this 

RelyingPartyRegistrations
                .fromMetadataLocation(oktaMetadataUrl)
                .registrationId("okta")).build();

Similar to that, it would be nice to have an option where we can load the relying party details when we have them in the database or some location already. We just need to load that file instead of adding the details one by one to the RelyingpartyRegistration.

it will be easier if I have an option where I have to mention the metadata location of the sp as well like this 
RelyingPartyRegistrations
                .fromMetadataLocationofSP(location)
                .fromMetadataLocationofIDP(oktaMetadataUrl)
                .registrationId("okta")).build();

**Comment by jzheaux** [comment]
Thanks for the suggestion, @sasirekha98. Are you interested in supplying a PR for this? You can look at the various `JdbcXXX` classes already in Spring Security for guidance.

**Comment by wapkch** [comment]
@jzheaux May i work on this?

**Comment by jzheaux** [comment]
Sorry that I missed you, @wapkch. Are you still interested in working on this? If so, please comment here and feel free also to proceed with a PR.

**Comment by wapkch** [comment]
Yes, I would be happy to work on this. I’ll proceed with the PR shortly.
